tensorzero-core/src/providers/tensorzero_relay.rs

```diff
@@ -0,0 +1,571 @@
+                Ok(chunk) => {
+                    let raw_chunk = serde_json::to_string(&chunk).unwrap_or_default();
+                    match chunk {
+                        InferenceResponseChunk::Chat(c) => Ok(ProviderInferenceResponseChunk::new(
+                            c.content,
+                            c.usage,
+                            // TODO - get the original chunk as a string
+                            raw_chunk,
+                            start_time.elapsed(),
+                            c.finish_reason,
+                        )),
+                        InferenceResponseChunk::Json(c) => Ok(ProviderInferenceResponseChunk::new(
+                            vec![ContentBlockChunk::Text(TextChunk {
+                                id: "0".to_string(),
+                                text: c.raw,
+                            })],
+                            c.usage,
+                            // TODO - get the original chunk as a string
+                            raw_chunk,
+                            start_time.elapsed(),
+                            c.finish_reason,
+                        )),
+                    }
+                }
+                Err(e) => Err(e),
+            })
+            .boxed()
+            .peekable();
+
+        Ok((stream, res.raw_request))
+    }
+
+    async fn start_batch_inference<'a>(
+        &'a self,
+        _requests: &'a [ModelInferenceRequest<'_>],
+        _client: &'a TensorzeroHttpClient,
+        _dynamic_api_keys: &'a InferenceCredentials,
+    ) -> Result<StartBatchProviderInferenceResponse, Error> {
+        Err(ErrorDetails::UnsupportedModelProviderForBatchInference {
+            provider_type: PROVIDER_TYPE.to_string(),
+        }
+        .into())
+    }
+
+    async fn poll_batch_inference<'a>(
+        &'a self,
+        _batch_request: &'a BatchRequestRow<'a>,
+        _http_client: &'a TensorzeroHttpClient,
+        _dynamic_api_keys: &'a InferenceCredentials,
+    ) -> Result<PollBatchInferenceResponse, Error> {
+        Err(ErrorDetails::UnsupportedModelProviderForBatchInference {
+            provider_type: PROVIDER_TYPE.to_string(),
+        }
+        .into())
+    }
+}
+
+#[derive(Clone, Debug)]
+pub enum TensorZeroRelayCredentials {
+    Static(SecretString),
+    Dynamic(String),
+    None,
+    WithFallback {
+        default: Box<TensorZeroRelayCredentials>,
+        fallback: Box<TensorZeroRelayCredentials>,
+    },
+}
+
+impl TryFrom<Credential> for TensorZeroRelayCredentials {
+    type Error = Error;
+
+    fn try_from(credentials: Credential) -> Result<Self, Error> {
+        match credentials {
+            Credential::Static(key) => Ok(TensorZeroRelayCredentials::Static(key)),
+            Credential::Dynamic(key_name) => Ok(TensorZeroRelayCredentials::Dynamic(key_name)),
+            Credential::None => Ok(TensorZeroRelayCredentials::None),
+            Credential::Missing => Ok(TensorZeroRelayCredentials::None),
```

Comment:
missing should probably throw?

Reply:
I don't think that's right - we use a `None` variant in other providers like vllm and sglang